contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -507,77 +699,86 @@ contract NodeOperatorsRegistry is AragonApp, IStakingModule, Versioned {
-        _auth(REQUEST_VALIDATORS_KEYS_FOR_DEPOSITS_ROLE);
+        _auth(STAKING_ROUTER_ROLE);
 
         uint256[] memory nodeOperatorIds;
         uint256[] memory activeKeysCountAfterAllocation;
         uint256[] memory exitedSigningKeysCount;
-        (
-            enqueuedValidatorsKeysCount,
-            nodeOperatorIds,
-            activeKeysCountAfterAllocation,
-            exitedSigningKeysCount
-        ) = _getSigningKeysAllocationData(_keysCount);
+        (enqueuedValidatorsKeysCount, nodeOperatorIds, activeKeysCountAfterAllocation, exitedSigningKeysCount) =
+            _getSigningKeysAllocationData(_keysCount);
 
         if (enqueuedValidatorsKeysCount == 0) {
             return (0, new bytes(0), new bytes(0));
         }
 
         (publicKeys, signatures) = _loadAllocatedSigningKeys(
-            enqueuedValidatorsKeysCount,
-            nodeOperatorIds,
-            activeKeysCountAfterAllocation,
-            exitedSigningKeysCount
+            enqueuedValidatorsKeysCount, nodeOperatorIds, activeKeysCountAfterAllocation, exitedSigningKeysCount
         );
 
-        SigningKeysStats.State memory totalSigningKeysStats = _getTotalSigningKeysStats();
-        totalSigningKeysStats.increaseDepositedSigningKeysCount(uint64(enqueuedValidatorsKeysCount));
-        _setTotalSigningKeysStats(totalSigningKeysStats);
+        Packed64x4.Packed memory totalSigningKeysStats = _loadTotalSigningKeysStats();
+        totalSigningKeysStats.inc(DEPOSITED_KEYS_COUNT_OFFSET, uint64(enqueuedValidatorsKeysCount));
+        _saveTotalSigningKeysStats(totalSigningKeysStats);
         _increaseValidatorsKeysNonce();
     }
 
+    function _getCorrectedNodeOperator(uint256 _nodeOperatorId)
+        internal
+        view
+        returns (uint64 vettedSigningKeysCount, uint64 exitedSigningKeysCount, uint64 depositedSigningKeysCount)
+    {
+        if (!getNodeOperatorIsActive(_nodeOperatorId)) return (0, 0, 0);
+
+        Packed64x4.Packed memory signingKeysStats = _loadOperatorSigningKeysStats(_nodeOperatorId);
+        vettedSigningKeysCount = signingKeysStats.get(VETTED_KEYS_COUNT_OFFSET);
+        exitedSigningKeysCount = signingKeysStats.get(EXITED_KEYS_COUNT_OFFSET);
+        depositedSigningKeysCount = signingKeysStats.get(DEPOSITED_KEYS_COUNT_OFFSET);
+
+        Packed64x4.Packed memory operatorTargetStats = _loadOperatorTargetValidtatorsStats(_nodeOperatorId);
+        // correct vetted count according to target
+        if (operatorTargetStats.get(TARGET_VALIDATORS_ACTIVE_OFFSET) > 0) {
+            // if target is enabled
+            uint64 targetValidatorsCount = operatorTargetStats.get(TARGET_VALIDATORS_COUNT_OFFSET);
+            if (targetValidatorsCount <= depositedSigningKeysCount.sub(exitedSigningKeysCount)) {
+                vettedSigningKeysCount = depositedSigningKeysCount;
+            } else {
+                vettedSigningKeysCount = Math64.min(vettedSigningKeysCount, exitedSigningKeysCount.add(targetValidatorsCount));
```

Comment:
why `exitedSigningKeysCount+targetValidatorsCount`? and not just `targetValidatorsCount`?

Reply:
it's correct. 
targetValidatorsCount - is about active keys